corehq/apps/data_cleaning/forms/cleaning.py

```diff
@@ -0,0 +1,164 @@
+
+        offcanvas_selector = "#offcanvas-bulk-changes"
+
+        alpine_data_model = {
+            "propId": initial_prop_id,
+            "cleanAction": self.data.get('clean_action', EditActionType.CHOICES[0][0]),
+            "findActions": [EditActionType.FIND_REPLACE],
+            "replaceActions": [EditActionType.REPLACE],
+            "copyActions": [EditActionType.COPY_REPLACE],
+        }
+
+        self.helper = FormHelper()
+        self.helper.form_tag = False
+        self.helper.layout = crispy.Layout(
+            crispy.Div(
+                crispy.Field(
+                    'clean_prop_id',
+                    x_select2=json.dumps({
+                        "placeholder": _("Select a case property..."),
+                        "dropdownParent": offcanvas_selector,
+                    }),
+                    **({
+                        "@select2change": "propId = $event.detail;",
+                    })
+                ),
+                crispy.Div(
+                    crispy.Field(
+                        'clean_action',
+                        x_select2=json.dumps({
+                            "placeholder": _("Select a cleaning action..."),
+                            "dropdownParent": offcanvas_selector,
+                        }),
+                        **({
+                            "@select2change": "cleanAction = $event.detail;",
+                        })
+                    ),
+                    crispy.Div(
+                        crispy.Div(
+                            'find_string',
+                            hqcrispy.CheckboxField('use_regex'),
+                            'replace_string',
+                            css_class="card-body",
+                        ),
+                        x_show="findActions.includes(cleanAction)",
+                        css_class="card mb-3",
+                    ),
+                    crispy.Div(
+                        crispy.Div(
+                            'replace_all_string',
+                            css_class="card-body",
+                        ),
+                        x_show="replaceActions.includes(cleanAction)",
+                        css_class="card mb-3",
+                    ),
+                    crispy.Div(
+                        crispy.Div(
+                            crispy.Field(
+                                'copy_from_prop_id',
+                                x_select2=json.dumps({
+                                    "placeholder": _("Select a case property..."),
+                                    "dropdownParent": offcanvas_selector,
+                                }),
+                            ),
+                            css_class="card-body",
+                        ),
+                        x_show="copyActions.includes(cleanAction)",
+                        css_class="card mb-3",
+                    ),
+                    twbscrispy.StrictButton(
```

Comment:
Will this eventually have enable/disable logic?

Reply:
it's not really needed, as the form will validate before it actually applies the changes. once the case property is selected there is no configuration that would enable/disable that can't be validated for (easier than enabling/disabling)

Reply:
What's the user experience like if the form validation fails?